visualization/app/codeCharta/core/settings/settings.service.ts

```diff
@@ -236,6 +237,63 @@ export class SettingsService implements DataServiceSubscriber, CameraChangeSubsc
 
     }
 
+    /**
+     * @returns {number}
+     *
+     * Function that computes the margin applied to a scenario related the square root of (the area divided
+     * by the number of buildings)
+     */
+    public  computeMargin(): number{
+
+        let margin: number;
+        if(this.dataService.data.renderMap != null && this.settings.dynamicMargin){
+            let  root: CodeMapNode = this.dataService.data.renderMap.root;
+
+            // Previous parameters are wrapped to introduce them into goThroughMap() and make value changes
+            // remain, with a pointer-like behaviour
+            let numberOfBuildingsWrapped = {numberOfBuildings: 0};
+            let areaParametersWrapped = {areaMetricName: this.settings.areaMetric, totalArea: 0};
+
+            this.dataService.goThroughMap( root, this.countBuildings, numberOfBuildingsWrapped);
+            this.dataService.goThroughMap( root, this.calculateTotalArea, areaParametersWrapped);
```

Comment:
I am not sure why, but for some projects the properties `numberOfBuildingsWrapped` and `areaParametersWrapped` have the value 0 and the `margin` just uses the default number of 15. For example I tested it with my own `codecharta.json` and `git.json` files, which were imported from the SCMLogParser.